Assets/MixedRealityToolkit.Extensions/Webrtc/Scripts/WebrtcPeerVideo.cs

```diff
@@ -0,0 +1,158 @@
+        /// TODO(bengreenier): This could be auto-bound with editor script
+        /// </summary>
+        /// <remarks>
+        /// This is expected to be bound to <see cref="Webrtc.OnInitialized"/> via the UnityEditor
+        /// </remarks>
+        public void OnInitialized()
+        {
+            // cache the peer.
+            this.peer = this.GetComponent<Webrtc>().Peer;
+
+            // setup the video players frame queue references.
+            this.LocalPlayer.FrameQueue = localFrameQueue;
+            this.RemotePlayer.FrameQueue = remoteFrameQueue;
+
+            // setup the peer events.
+            this.peer.LocalI420FrameReady += Peer_LocalI420FrameReady;
+            this.peer.RemoteI420FrameReady += Peer_RemoteI420FrameReady;
+        }
+
+        /// <summary>
+        /// Unity Engine OnDestroy() hook
+        /// </summary>
+        /// <remarks>
+        /// https://docs.unity3d.com/ScriptReference/MonoBehaviour.OnDestroy.html
+        /// </remarks>
+        private void OnDestroy()
+        {
+            // cleanup, if necessary
+            if (this.peer != null)
+            {
+                // cleanup the peer events.
+                this.peer.LocalI420FrameReady -= Peer_LocalI420FrameReady;
+                this.peer.RemoteI420FrameReady -= Peer_RemoteI420FrameReady;
+
+                // cleanup the peer.
+                this.peer = null;
+            }
+        }
+
+        private void Peer_LocalI420FrameReady(IntPtr dataY, IntPtr dataU, IntPtr dataV, IntPtr dataA, int strideY, int strideU, int strideV, int strideA, uint width, uint height)
+        {
+            FramePacket packet = localFrameQueue.GetDataBufferWithoutContents((int)(width * height * 4));
+            if (packet == null)
+            {
+                return;
+            }
+            CopyYuvToBuffer(dataY, dataU, dataV, strideY, strideU, strideV, width, height, packet.Buffer);
+            packet.width = (int)width;
+            packet.height = (int)height;
+            localFrameQueue.Push(packet);
+        }
+
+        private void Peer_RemoteI420FrameReady(IntPtr dataY, IntPtr dataU, IntPtr dataV, IntPtr dataA, int strideY, int strideU, int strideV, int strideA, uint width, uint height)
+        {
+            FramePacket packet = remoteFrameQueue.GetDataBufferWithoutContents((int)(width * height * 4));
+            if (packet == null)
+            {
+                return;
+            }
+            CopyYuvToBuffer(dataY, dataU, dataV, strideY, strideU, strideV, width, height, packet.Buffer);
+            packet.width = (int)width;
+            packet.height = (int)height;
+            remoteFrameQueue.Push(packet);
+        }
+
+        /// <summary>
+        /// Internal yuv data copy routine
+        /// </summary>
+        private void CopyYuvToBuffer(IntPtr dataY, IntPtr dataU, IntPtr dataV,
+        int strideY, int strideU, int strideV,
+        uint width, uint height, byte[] buffer)
+        {
+            unsafe
```

Comment:
>unsafe [](start = 12, length = 6)

unsafe code can't always be adopted by community members. Do you know if adding this script to the mrtk and opening one of the mrtk demos requires allowing unsafe code?

Reply:
well you've gone and stumped me - because normally i'd agree and expect to see "Allow unsafe code" in player settings would be needed for this to work at all, in the included Example, as well as in other ones (basically anywhere this script is built..) but i'm not seeing that behavior locally...

I've got it unchecked and am able to run `mono` and `il2cpp` backend-based projects, in editor and in build (standalone). perhaps somehow related to this method being private?

Reply:
It's generally located as a checkbox in the assembly definition

Reply:
🤦‍♂️ @StephenHodgson, oh wow, silly me - thanks for calling that out!

@chrisfromwork i'm not sure how to best validate this...here's what i tried:

il2cpp builds of `BoundryVisualization` and `WebrtcBasicSignalExample` seem to both include this code (ctrl+f in generated vs solution) - so i would guess that unsafe flag is always needed when building scripts from my asmdef, which seem to be built in all scenarios.